specification/search/data-plane/Azure.Search/preview/2025-05-01-preview/searchservice.json

```diff
@@ -0,0 +1,14715 @@
+    },
+    "VectorSearchAlgorithmKind": {
+      "type": "string",
+      "enum": [
+        "hnsw",
+        "exhaustiveKnn"
+      ],
+      "x-ms-enum": {
+        "name": "VectorSearchAlgorithmKind",
+        "modelAsString": true,
+        "values": [
+          {
+            "value": "hnsw",
+            "name": "Hnsw",
+            "description": "HNSW (Hierarchical Navigable Small World), a type of approximate nearest neighbors algorithm."
+          },
+          {
+            "value": "exhaustiveKnn",
+            "name": "ExhaustiveKnn",
+            "description": "Exhaustive KNN algorithm which will perform brute-force search."
+          }
+        ]
+      },
+      "description": "The algorithm used for indexing and querying."
+    },
+    "VectorSearchCompressionConfiguration": {
+      "type": "object",
+      "discriminator": "kind",
+      "properties": {
+        "name": {
+          "externalDocs": {
+            "url": "https://learn.microsoft.com/rest/api/searchservice/Naming-rules"
+          },
+          "type": "string",
+          "description": "The name to associate with this particular configuration.",
+          "x-nullable": false
+        },
+        "kind": {
+          "$ref": "#/definitions/VectorSearchCompressionKind",
+          "description": "The name of the kind of compression method being configured for use with vector search.",
+          "x-nullable": false
+        },
+        "rerankWithOriginalVectors": {
+          "type": "boolean",
+          "description": "If set to true, once the ordered set of results calculated using compressed vectors are obtained, they will be reranked again by recalculating the full-precision similarity scores. This will improve recall at the expense of latency.",
+          "x-nullable": true
+        },
+        "defaultOversampling": {
+          "type": "number",
+          "format": "double",
+          "description": "Default oversampling factor. Oversampling will internally request more documents (specified by this multiplier) in the initial search. This increases the set of results that will be reranked using recomputed similarity scores from full-precision vectors. Minimum value is 1, meaning no oversampling (1x). This parameter can only be set when rerankWithOriginalVectors is true. Higher values improve recall at the expense of latency.",
+          "x-nullable": true
+        },
+        "rescoringOptions": {
+          "type": "object",
+          "$ref": "#/definitions/RescoringOptions",
+          "description": "Contains the options for rescoring.",
+          "x-nullable": true
+        },
+        "truncationDimension": {
+          "type": "integer",
+          "format": "int32",
+          "description": "The number of dimensions to truncate the vectors to. Truncating the vectors reduces the size of the vectors and the amount of data that needs to be transferred during search. This can save storage cost and improve search performa
```

Comment:
This might be pedantic, but would it ever reasonably be possible for int32 to be insufficient for the number of documents processed?